### PR TITLE
[Fix CI ES Test] Fix Search Flaky Tests

### PR DESCRIPTION
**Pull request by mohityadav766**
- Improvement for Search Tests [In Progress]

### Describe your changes:

Fixes <issue-number>

I worked on ... because ...

#
### Type of change:

- [x] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/search/SearchClient.java [hunk 1]
```diff
@@ -79,6 +79,8 @@ public interface SearchClient {
 
   Response search(SearchRequest request) throws IOException;
 
+  Response getDocWithId(String indexName, String entityId) throws IOException;
```
Comment (TeddyCr) [inline openmetadata-service/src/main/java/org/openmetadata/service/search/SearchClient.java:82]: should we rename this getDocByID?

File: openmetadata-service/src/test/java/org/openmetadata/service/resources/EntityResourceTest.java [hunk 2]
```diff
@@ -2223,59 +2224,69 @@ protected void deleteTagAndCheckRelationshipsInSearch(TestInfo test)
...
+    AtomicReference<Map<String, Object>> responseMap = new AtomicReference<>();
+    Awaitility.await("Wait for Indexes to be updated for the Entity")
+        .pollInterval(Duration.ofMillis(2000L))
+        .atMost(Duration.ofMillis(120 * 1000L)) // 60 iterations for 120 seconds
+        .until(
+            () -> {
+              responseMap.set(getEntityDocumentFromSearch(entityId, entityType));
+              Object esLastUpdateAt = responseMap.get().get("updatedAt");
+              long esLastUpdate = Long.parseLong(esLastUpdateAt.toString());
+              return esLastUpdate >= entityDbLastUpdate;
```
Comment (TeddyCr) [inline openmetadata-service/src/test/java/org/openmetadata/service/resources/EntityResourceTest.java:2264]: Do you think we could have scenarios where 1 doc in he index evaluates the condition to true but not all the doc we'll need to compare are yet indexed?

Comment (mohityadav766) [reply openmetadata-service/src/test/java/org/openmetadata/service/resources/EntityResourceTest.java:2264]: correct @TeddyCr , this is in Progress actually , need to do some more work

File: openmetadata-service/src/main/java/org/openmetadata/service/search/SearchClient.java (hunk 1 above)
Comment (mohityadav766) [reply openmetadata-service/src/main/java/org/openmetadata/service/search/SearchClient.java]: done

**Review by harshach — APPROVED** [review]

**Pull request merged**